addons/mail/models/discuss/discuss_channel.py

```diff
@@ -160,6 +161,22 @@ def _constraint_group_id_channel(self):
 
     # COMPUTE / INVERSE
 
+    @api.depends("channel_member_ids", "auto_recompute_name")
+    def _compute_name(self):
+        for channel in self:
+            if channel.auto_recompute_name:
```

Comment:
could gain indent level in whole function by pre-filtering the channels to loop:

```suggestion
        for channel in self.filtered(...):
```